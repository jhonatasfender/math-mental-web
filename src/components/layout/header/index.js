import styled from 'styled-components';
import { DarkTheme } from 'styled-icons/fluentui-system-filled';

const HeaderStyle = styled.header`
  display: flex;
`;

const Dark = styled(DarkTheme)`
  width: 3rem;
  grid-area: header;
`;

const Header = ({ onClick }) => (
  <HeaderStyle>
    <Dark
      onClick={onClick}
      type="button"
      role="button"
      aria-hidden="false"
      aria-label="dark mode"
    />
  </HeaderStyle>
);
export default Header;
