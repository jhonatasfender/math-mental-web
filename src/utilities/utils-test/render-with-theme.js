import Main from '@components/layout/main';
import { render } from '@testing-library/react';

export const renderWithMain = (children) => render(<Main>{children}</Main>);
