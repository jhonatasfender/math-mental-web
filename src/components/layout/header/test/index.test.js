import Header from '@components/layout/header';
import { renderWithMain } from '@utils-test/render-with-theme';

describe('<Header />', () => {
  it('should render', () => {
    const { container } = renderWithMain(<Header />);

    expect(container).toMatchSnapshot();
  });
});
