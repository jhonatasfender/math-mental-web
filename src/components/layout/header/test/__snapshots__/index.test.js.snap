// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render 1`] = `
.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  width: 3rem;
  grid-area: header;
}

.c4 {
  grid-area: body;
  padding: 1rem 17rem;
  background: #EDEDED;
  overflow: auto;
}

.c0 {
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 5.3rem 1.8fr;
  gap: 0px 0px;
  grid-template-areas: 'header' 'body';
}

<div>
  <main
    class="c0"
  >
    <header
      class="c1"
    >
      <svg
        aria-hidden="false"
        aria-label="dark mode"
        class="c2 c3"
        fill="currentColor"
        focusable="false"
        role="button"
        type="button"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 22a10 10 0 1 0 0-20 10 10 0 0 0 0 20zm0-2V4a8 8 0 1 1 0 16z"
        />
      </svg>
    </header>
    <div
      class="c4"
    >
      <header
        class="c1"
      >
        <svg
          aria-hidden="false"
          aria-label="dark mode"
          class="c2 c3"
          fill="currentColor"
          focusable="false"
          role="button"
          type="button"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 22a10 10 0 1 0 0-20 10 10 0 0 0 0 20zm0-2V4a8 8 0 1 1 0 16z"
          />
        </svg>
      </header>
    </div>
  </main>
</div>
`;
